src/main/java/de/fraunhofer/aisec/cpg/graph/NamespaceDeclaration.java

```diff
@@ -26,4 +26,104 @@
 
 package de.fraunhofer.aisec.cpg.graph;
 
-public class NamespaceDeclaration extends TranslationUnitDeclaration {}
+import de.fraunhofer.aisec.cpg.helpers.NodeComparator;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+/**
+ * Declares the scope of a namespace and appends its own name to the current namespace-ürefix to
+ * form a new namespace prefix. While RecordDeclarations in C++ and Java have their own namespace,
+ * namespace declarations can be declared multiple times. At the beginning of a Java-file, a
+ * namespace declaration is used to represent the package name as namespace. In its explicit
+ * appearance a namespace declaration can contain {@link FunctionDeclaration}, {@link
+ * FieldDeclaration} and {@link RecordDeclaration} similar to a {@link RecordDeclaration} and the
+ * semantical difference between NamespaceDeclaration and {@link RecordDeclaration} lies in the
+ * non-instantiabillity of a namespace.
```

Comment:
For me, a Translation unit is clearly only for a file, or a piece of code, so I do see a Namespace close to a `RecordDeclaration` especially because of RecordDeclarations adding "qualification" to a type name. As mentioned, the difference lies in the non-instantiabillity, which means that a Namespace can only have the static variants of fields and functions.